File: tags/alpha-candidate/common/game_server/src/sc/server/network/AuthenticationFailedException.java
package sc.server.network;

import sc.api.plugins.exceptions.RescueableClientException;

public class AuthenticationFailedException extends RescueableClientException
{
	/**
	 * 
	 */
	private static final long	serialVersionUID	= 8849650876076674212L;

	public AuthenticationFailedException()
	{
		super("Failed to authenticate as administrator");
	}
}

File: server/src/sc/server/network/AuthenticationFailedException.kt
package sc.server.network

import sc.api.plugins.exceptions.RescuableClientException
import sc.protocol.requests.AdminLobbyRequest

/** Thrown if attempting to authenticate with a wrong password. */
class AuthenticationFailedException:
        RescuableClientException("Failed to authenticate as administrator")

class UnauthenticatedException(packet: AdminLobbyRequest):
        RescuableClientException("Request $packet requires authentication")